---
title: "MT-exp1-Pande"
author: "Etienne Yergeau"
date: "May 2, 2019"
output: html_document
---



```{r libraries}
install.packages("pacman")
pacman::p_load(pacman, EBSeq, ggpubr, tidyverse, RColorBrewer)
```



```{r import data and prepare}
#Import files
MT <- read.table(file = "../data/raw/merged_gene_abundance.tsv", row.names = 1, header = T, comment.char = "", sep = "\t") # 1 269 055 obs in 47 var
map <- read.table(file = "../data/raw/mapping_file.tsv", row.names = 1, header = T, sep = "\t", comment.char = "") #47 obs in 3 var
annot <- read.table(file = "../data/raw/annotations.tsv", header = T, sep = "\t", comment.char = "", quote = "") #1269056 obs in 33 vars
SWC <- read.table(file = "../data/raw/SWC_DT.txt", header = T, sep = "\t") # 24 obs in 5 vars

#prepare files
MT.t <- t(MT)
#Sort everything
map.s <- map[order(row.names(map)),]
MT.s <- MT.t[order(row.names(MT.t)),]
row.names(MT.s) == row.names(map.s)#Sanity check -- OK

#Remove large intermediate objects to free memory
rm(MT)
rm(MT.t)

```


```{r differential abundance-RHIZO}
#EB test from EB seq for multiple testing of hypothesis

#rhizo 25 vs.100
map.s.rhizo <- map.s[(map.s$Sample == "Rhizosphere") & (map.s$WHC == "25" | map.s$WHC == "100"), ] # 12 obs in 3 var
MT.rhizo <- MT.s[(map.s$Sample == "Rhizosphere") & (map.s$WHC == "25" | map.s$WHC == "100"), ]

#Data file should be genes in rows and samples in columns
MT.rhizo <- t(MT.rhizo)

#Remove rows with all zeroes
length(which(rowSums(MT.rhizo) == 0)) #75 554 out of 1269055. Should have 1 193 501 left
MT.rhizo <- MT.rhizo[!rowSums(MT.rhizo) == 0,]
dim(MT.rhizo) #OK: 1193501   12

#Create group file L = 25, H = 100
SWC.rhizo = as.factor(c("H", "L","H", "L","H", "L","H", "L", "L", "H", "L", "H"))

#Create size Factors file
size.rhizo = MedianNorm(MT.rhizo)

#Run EBTest
EBtestOUT.rhizo <- EBTest(MT.rhizo, Conditions = SWC.rhizo, sizeFactors = size.rhizo, maxround = 5)

#Get results
DE.rhizo <- GetDEResults(EBtestOUT.rhizo, FDR = 0.05)

#Calculate log2 fold change and mean counts
fold.rhizo <- data.frame(row.names = row.names(MT.rhizo))
fold.rhizo$mean <- rowMeans(MT.rhizo)
fold.rhizo$log2 <- log2(rowMeans(MT.rhizo[,SWC.rhizo == "L"])/rowMeans(MT.rhizo[,SWC.rhizo == "H"]))

#Get annotations for DE at 0.05
rhizo.DE.annot <- annot[annot$gene_id %in% DE.rhizo$DEfound,]

#Get annotations for DE
DE.bact.rhizo <- rhizo.DE.annot[rhizo.DE.annot$tax_kingdom == "Bacteria",2]
bact.point.rhizo <- fold.rhizo[row.names(fold.rhizo) %in% DE.bact.rhizo,]
DE.arch.rhizo <- rhizo.DE.annot[rhizo.DE.annot$tax_kingdom == "Archaea",2]
arch.point.rhizo <- fold.rhizo[row.names(fold.rhizo) %in% DE.arch.rhizo,]
DE.fun.rhizo <- rhizo.DE.annot[grep("mycota", rhizo.DE.annot$tax_phylum, ignore.case = TRUE),2]
fun.point.rhizo <- fold.rhizo[row.names(fold.rhizo) %in% DE.fun.rhizo,]
DE.plant.rhizo <- rhizo.DE.annot[rhizo.DE.annot$tax_phylum == "Streptophyta",2]
plant.point.rhizo <- fold.rhizo[row.names(fold.rhizo) %in% DE.plant.rhizo,]
DE.null.rhizo <- rhizo.DE.annot[rhizo.DE.annot$tax_kingdom == "NULL",2]
null.point.rhizo <- fold.rhizo[row.names(fold.rhizo) %in% DE.null.rhizo,]
DE.other.rhizo <- rhizo.DE.annot[!rhizo.DE.annot$gene_id %in% append(append(DE.bact.rhizo,DE.fun.rhizo),append(DE.plant.rhizo, append(DE.null.rhizo,DE.arch.rhizo))),2]
other.point.rhizo <- fold.rhizo[row.names(fold.rhizo) %in% DE.other.rhizo,]

#Plots
#Only plot first 100 000 points for clarity
volcano.rhizo <- ggplot() +
 geom_point(data = fold.rhizo[1:100000,], aes(x = mean, y = log2)) +
 geom_point(data = other.point.rhizo, aes(x = mean, y = log2), color = "yellow") +
 geom_point(data = null.point.rhizo, aes(x = mean, y = log2), color = "purple") +
 geom_point(data = bact.point.rhizo, aes(x = mean, y = log2), color = "red") +
 geom_point(data = fun.point.rhizo, aes(x = mean, y = log2), color = "blue") +
 geom_point(data = arch.point.rhizo, aes(x = mean, y = log2), color = "pink") +
 geom_point(data = plant.point.rhizo, aes(x = mean, y = log2), color = "green") +
 geom_hline(yintercept = 0, linetype = "solid") +
 xlab("Mean read count") + 
 ylab("Log 2 fold change (25%/100%)") + 
 scale_x_log10() +
 theme_bw()
volcano.rhizo 

#Sort for exporting
rhizo.DE.annot <- rhizo.DE.annot[order(rhizo.DE.annot$gene_id),]
DEfound.rhizo <- data.frame(DE.rhizo$DEfound)
DEfound.rhizo <- DEfound.rhizo[order(DEfound.rhizo$DE.rhizo.DEfound),]
fold.rhizo.DE <- fold.rhizo[row.names(fold.rhizo) %in% DE.rhizo$DEfound,]
fold.rhizo.DE <- fold.rhizo.DE[order(row.names(fold.rhizo.DE)),]
pval.rhizo.DE <- data.frame(DE.rhizo$PPMat[row.names(DE.rhizo$PPMat) %in% DE.rhizo$DEfound,])
pval.rhizo.DE <- pval.rhizo.DE[order(row.names(pval.rhizo.DE)),]
sum(rhizo.DE.annot$gene_id == DEfound.rhizo) #Sanity check, should be 21765
sum(row.names(fold.rhizo.DE) == DEfound.rhizo) #Sanity check, should be 21765
sum(row.names(pval.rhizo.DE) == DEfound.rhizo)#Sanity check, should be 21765
DEfound.rhizo <- cbind(DEfound.rhizo,fold.rhizo.DE, pval.rhizo.DE, rhizo.DE.annot)

#Export
#Matrix of all genes with DE and EE p-values
write.table(as.matrix(DE.rhizo$PPMat), "../output/PPmat_rhizo.txt", eol = "\n", sep = "\t")
#List of the DE transcripts, p-Values, annotations, abundance
write.table(DEfound.rhizo, "../output/DEfound_rhizo.txt", eol = "\n", sep = "\t")
#Status of the transcript: DE, EE, or Filtered+Reason
write.table(as.matrix(DE.rhizo$Status), "../output/Status_rhizo.txt", eol = "\n", sep = "\t")
```

```{r differential abundance-ROOT}
#EB test from EB seq for multiple testing of hypothesis

#root 25 vs.100
map.s.root <- map.s[(map.s$Sample == "Root") & (map.s$WHC == "25" | map.s$WHC == "100"), ]
MT.root <- MT.s[(map.s$Sample == "Root") & (map.s$WHC == "25" | map.s$WHC == "100"), ]

#Data file should be genes in lines and samples in columns
MT.root <- t(MT.root)

#Remove lines with all zeroes
length(which(rowSums(MT.root) == 0)) #132 468 out of 1269055. Should have 1 136 587 left
MT.root <- MT.root[!rowSums(MT.root) == 0,]
dim(MT.root) #OK: 1 136 587   11

#Create group file L = 25%, H = 100%
SWC.root <- as.factor(c("L", "L","H", "L","H", "H","L", "H", "H", "H", "L"))

#Create size Factors file
size.root = MedianNorm(MT.root)

#Run EBTest
EBtestOUT.root <- EBTest(MT.root, Conditions = SWC.root, sizeFactors = size.root, maxround = 5)

#Get results
DE.root <- GetDEResults(EBtestOUT.root, FDR = 0.05)

#Calculate log2 fold change and mean counts
fold.root <- data.frame(row.names = row.names(MT.root))
fold.root$mean <- rowMeans(MT.root)
fold.root$log2 <- log2(rowMeans(MT.root[,SWC.root == "L"])/rowMeans(MT.root[,SWC.root == "H"]))

#Get annotations for DE at 0.05
root.DE.annot <- annot[annot$gene_id %in% DE.root$DEfound,]

#Get annotations for DE
DE.bact.root <- root.DE.annot[root.DE.annot$tax_kingdom == "Bacteria",2]
bact.point.root <- fold.root[row.names(fold.root) %in% DE.bact.root,]
DE.fun.root <- root.DE.annot[grep("mycota", root.DE.annot$tax_phylum, ignore.case = TRUE),2]
fun.point.root <- fold.root[row.names(fold.root) %in% DE.fun.root,]
DE.plant.root <- root.DE.annot[root.DE.annot$tax_phylum == "Streptophyta",2]
plant.point.root <- fold.root[row.names(fold.root) %in% DE.plant.root,]
DE.null.root <- root.DE.annot[root.DE.annot$tax_kingdom == "NULL",2]
null.point.root <- fold.root[row.names(fold.root) %in% DE.null.root,]
DE.other.root <- root.DE.annot[!root.DE.annot$gene_id %in% append(append(DE.bact.root,DE.fun.root),append(DE.plant.root, DE.null.root)),2]
other.point.root <- fold.root[row.names(fold.root) %in% DE.other.root,]

#Plots
#Only plot first 100 000 for clarity
volcano.root <- ggplot() +
 geom_point(data = fold.root[1:100000,], aes(x = mean, y = log2)) +
 geom_point(data = other.point.root, aes(x = mean, y = log2), color = "yellow") +
 geom_point(data = null.point.root, aes(x = mean, y = log2), color = "purple") +
 geom_point(data = bact.point.root, aes(x = mean, y = log2), color = "red") +
 geom_point(data = fun.point.root, aes(x = mean, y = log2), color = "blue") +
 geom_point(data = plant.point.root, aes(x = mean, y = log2), color = "green") +
 geom_hline(yintercept = 0, linetype = "solid") +
 xlab("Mean read count") + 
 ylab("Log 2 fold change (25%/100%)") + 
 scale_x_log10() +
 theme_bw()
volcano.root 


#Sort for exporting
root.DE.annot <- root.DE.annot[order(root.DE.annot$gene_id),]
DEfound.root <- data.frame(DE.root$DEfound)
DEfound.root <- DEfound.root[order(DEfound.root$DE.root.DEfound),]
fold.root.DE <- fold.root[row.names(fold.root) %in% DE.root$DEfound,]
fold.root.DE <- fold.root.DE[order(row.names(fold.root.DE)),]
pval.root.DE <- data.frame(DE.root$PPMat[row.names(DE.root$PPMat) %in% DE.root$DEfound,])
pval.root.DE <- pval.root.DE[order(row.names(pval.root.DE)),]
sum(root.DE.annot$gene_id == DEfound.root) #Sanity check, should be 42001
sum(row.names(fold.root.DE) == DEfound.root) #Sanity check, should be 42001
sum(row.names(pval.root.DE) == DEfound.root) #Sanity check, should be 42001
DEfound.root <- cbind(DEfound.root,fold.root.DE, pval.root.DE, root.DE.annot)

#Export
#Matrix of all genes with DE and EE p-values
write.table(as.matrix(DE.root$PPMat), "../output/PPmat_root.txt", eol = "\n", sep = "\t")
#List of the DE transcripts with P-value and annotations
write.table(DEfound.root, "../output/DEfound_root.txt", eol = "\n", sep = "\t")
#Status of the transcript: DE, EE, or Filtered+Reason
write.table(as.matrix(DE.root$Status), "../output/Status_root.txt", eol = "\n", sep = "\t")
```


```{r various stats Root}
#How many DE gene positive, negative?
length(fun.point.root[fun.point.root$log2 > 0,1]) #23042
length(fun.point.root[fun.point.root$log2 < 0,1]) #232
length(plant.point.root[plant.point.root$log2 > 0,1]) #1295
length(plant.point.root[plant.point.root$log2 < 0,1]) #3061
length(bact.point.root[bact.point.root$log2 > 0,1]) #2231
length(bact.point.root[bact.point.root$log2 < 0,1]) #78
length(other.point.root[other.point.root$log2 > 0,1]) #5895
length(other.point.root[other.point.root$log2 < 0,1]) #863

#How many total transcripts for each taxa ?
length(annot[(annot$tax_kingdom == "Bacteria") & (annot$gene_id %in% row.names(MT.root)),2]) #365 435
length(annot[(annot$tax_kingdom == "Archaea") & (annot$gene_id %in% row.names(MT.root)),2]) #12 792
length(annot[(annot$tax_kingdom == "NULL") & (annot$gene_id %in% row.names(MT.root)),2]) #221 331
length(annot[(annot$tax_kingdom == "Eukaryota") & (annot$gene_id %in% row.names(MT.root)),2]) #533 369
length(annot[(annot$tax_kingdom == "Viruses") & (annot$gene_id %in% row.names(MT.root)),2]) #3 660
length(annot[(annot$tax_phylum == "Streptophyta") & (annot$gene_id %in% row.names(MT.root)),2]) #200 823
length(annot[(annot$tax_phylum == "Ascomycota") & (annot$gene_id %in% row.names(MT.root)),2]) #151 695
length(annot[(annot$tax_phylum == "Basidiomycota") & (annot$gene_id %in% row.names(MT.root)),2]) #46 352
length(annot[(annot$tax_phylum == "Mucoromycota") & (annot$gene_id %in% row.names(MT.root)),2]) #2 186
(533369 - 200823 - 151695 - 46352 - 2186) #132 313

```

```{r various stats Rhizo}
#How many DE gene positive, negative?
length(fun.point.rhizo[fun.point.rhizo$log2 > 0,1]) #149
length(fun.point.rhizo[fun.point.rhizo$log2 < 0,1]) #253
length(plant.point.rhizo[plant.point.rhizo$log2 > 0,1]) #41
length(plant.point.rhizo[plant.point.rhizo$log2 < 0,1]) #178
length(bact.point.rhizo[bact.point.rhizo$log2 > 0,1]) #7938
length(bact.point.rhizo[bact.point.rhizo$log2 < 0,1]) #6240
length(other.point.rhizo[other.point.rhizo$log2 > 0,1]) #391
length(other.point.rhizo[other.point.rhizo$log2 < 0,1]) #1351

#How many total transcripts for each taxa ?
length(annot[(annot$tax_kingdom == "Bacteria") & (annot$gene_id %in% row.names(MT.rhizo)),2]) #430 984
length(annot[(annot$tax_kingdom == "Archaea") & (annot$gene_id %in% row.names(MT.rhizo)),2]) #14 943
length(annot[(annot$tax_phylum == "Streptophyta") & (annot$gene_id %in% row.names(MT.rhizo)),2]) #169 788
length(annot[(annot$tax_phylum == "Ascomycota") & (annot$gene_id %in% row.names(MT.rhizo)),2]) #140 046
length(annot[(annot$tax_phylum == "Basidiomycota") & (annot$gene_id %in% row.names(MT.rhizo)),2]) #44 485
length(annot[(annot$tax_phylum == "Mucoromycota") & (annot$gene_id %in% row.names(MT.rhizo)),2]) #2 214
length(annot[(annot$tax_kingdom == "NULL") & (annot$gene_id %in% row.names(MT.rhizo)),2]) #254 744
length(annot[(annot$tax_kingdom == "Viruses") & (annot$gene_id %in% row.names(MT.rhizo)),2]) #4 255
length(annot[(annot$tax_kingdom == "Eukaryota") & (annot$gene_id %in% row.names(MT.rhizo)),2]) #488 575
(488575 - 169788 - 140046 - 44485 - 2214) #132 042
```

```{r Plot various stats}
stats.all <- data.frame(
 Taxa = c("Archaea", "Bacteria", "Fungi", "Plants", "Viruses", "unclassified", "Others","Archaea", "Bacteria", "Fungi", "Plants", "Viruses", "unclassified", "Others"),
 Comp = c("Root","Root","Root","Root","Root","Root","Root","Rhizo","Rhizo","Rhizo","Rhizo","Rhizo","Rhizo","Rhizo"),
 Count = c(12792,365435,200233,200823,3660,221331,132313,14943, 430984, 186745,169788, 4255, 254744, 132042)
)
stats.DE <- data.frame(
 Taxa = c("Bacteria-More", "Bacteria-Less","Fungi-More", "Fungi-Less","Plants-More", "Plants-Less","unclassified", "Others","Bacteria-More", "Bacteria-Less","Fungi-More", "Fungi-Less","Plants-More", "Plants-Less","unclassified", "Others"),
 Comp = c("Root","Root","Root","Root","Root","Root","Root","Root", "Rhizo","Rhizo","Rhizo","Rhizo","Rhizo","Rhizo","Rhizo", "Rhizo"),
 Count = c(2231,78,23042,232,1295,3061,5303,6758,7938,6240,149,253,41,178,5224,1583)
)

palette(c("cyan", "red", "blue", "yellow", "green", "purple", "grey"))
stack.stats.all <- ggplot(stats.all, aes(fill = Taxa, y = Count, x = Comp)) + 
  geom_bar( stat = "identity", position = "fill") +
  ylab("Fraction of reads") + 
  scale_fill_manual(values = palette(), guide = guide_legend(label.theme = element_text(face = "italic", size = 8))) +
  theme_bw() +
  scale_y_continuous( expand = c(0,0)) +
  scale_x_discrete(name = "Compartment") +
  theme(axis.text.x = element_text(size = 8, angle = 45, hjust = 1))
stack.stats.all

palette(c("darkred", "red","darkblue", "blue", "yellow","darkgreen", "green", "purple"))
stack.stats.DE <- ggplot(stats.DE, aes(fill = Taxa, y = Count, x = Comp)) + 
  geom_bar( stat = "identity", position = "fill") +
  ylab("Fraction of reads") + 
  scale_fill_manual(values = palette(), guide = guide_legend(label.theme = element_text(face = "italic", size = 8))) +
  theme_bw() +
  scale_y_continuous( expand = c(0,0)) +
  scale_x_discrete(name = "Compartment") +
  theme(axis.text.x = element_text(size = 8, angle = 45, hjust = 1))
stack.stats.DE  

```


```{r stack barchart taxa Rhizo}
#Prokaryotes at phylum
#Create data frame
DEfound.rhizo.bact <- DEfound.rhizo[DEfound.rhizo$tax_kingdom == "Bacteria",]
phylum.table.rhizo = DEfound.rhizo.bact %>%
 group_by(tax_phylum) %>%
 summarise(
  n = n(),
  over = sum(log2 > 0, na.rm = TRUE),
  under = sum(log2 < 0, na.rm = TRUE),
  
 ) %>%
 filter(
  n > 100
 )
phylum.table.rhizo <- phylum.table.rhizo[-9,] #Remove NULL
row.names(phylum.table.rhizo) <- phylum.table.rhizo$tax_phylum
others.phylum.rhizo <- data.frame("Others", (14178 - sum(phylum.table.rhizo$n)), (7938 - sum(phylum.table.rhizo$over)), (6240 - sum(phylum.table.rhizo$under)))
colnames(others.phylum.rhizo) <- colnames(phylum.table.rhizo)
row.names(others.phylum.rhizo) <- "Others"
phylum.table.rhizo <- rbind(phylum.table.rhizo, others.phylum.rhizo) 
colSums(phylum.table.rhizo[,2:4]) #Should be 14178

#Get a column for all transcripts
annot.bact <- annot[annot$tax_kingdom == "Bacteria",]
annot.bact.rhizo <- annot.bact[annot.bact$gene_id %in% row.names(MT.rhizo),]
MT.rhizo.bact <- MT.rhizo[row.names(MT.rhizo) %in% annot.bact$gene_id,]
annot.bact.rhizo <- annot.bact.rhizo[order(annot.bact.rhizo$gene_id),]
MT.rhizo.bact <- MT.rhizo.bact[order(row.names(MT.rhizo.bact)),]
sum(row.names(MT.rhizo.bact) == annot.bact.rhizo$gene_id) #Sanity check
MT.rhizo.bact.tax <- data.frame(cbind(MT.rhizo.bact, annot.bact.rhizo$tax_phylum))

all.column.rhizo <- MT.rhizo.bact.tax %>%
 group_by(V13) %>%
 summarise(
  count = n(),
  
  
 )
all.column.rhizo <- all.column.rhizo[all.column.rhizo$V13 %in% phylum.table.rhizo$tax_phylum,]
row.names(all.column.rhizo) <- all.column.rhizo$V13
others.all.rhizo <- data.frame("Others", (430984 - sum(all.column.rhizo$count)))
colnames(others.all.rhizo) <- colnames(all.column.rhizo)
row.names(others.all.rhizo) <- "Others"
all.column.rhizo <- rbind(all.column.rhizo, others.all.rhizo) 
sum(all.column.rhizo$count)

all.column.rhizo$V13 == phylum.table.rhizo$tax_phylum #Sanity check
phylum.table.rhizo <- cbind(phylum.table.rhizo,all.column.rhizo$count)
colnames(phylum.table.rhizo) <- c("Phylum", "n", "More", "Less", "All")

#Prepare for ggplot
phylum.table.rhizo.long <- gather(phylum.table.rhizo,subset,counts,3:5) #transform in long format for ggplot
palette(c(brewer.pal(n = 9, name = "Set1"),"lightgrey", "black", "darkred", "darkblue", "darkgreen", "purple4", "darkgrey", "white"))
stack.phyla.rhizo <- ggplot(phylum.table.rhizo.long, aes(fill = Phylum, y = counts, x = subset)) + 
  geom_bar( stat = "identity", position = "fill") +
  ylab("Fraction of reads") + 
  scale_fill_manual(values = palette(), guide = guide_legend(label.theme = element_text(face = "italic", size = 8))) +
  theme_bw() +
  scale_y_continuous( expand = c(0,0)) +
  scale_x_discrete(name = "Subset") +
  theme(axis.text.x = element_text(size = 8, angle = 45, hjust = 1))
stack.phyla.rhizo
```

```{r stack barchart taxa root}
#Fungi at class
#Create data frame
DEfound.root.fun <- DEfound.root[grep(".mycota", DEfound.root$tax_phylum),]
class.table.root <- DEfound.root.fun %>%
 group_by(tax_class) %>%
 summarise(
  n = n(),
  over = sum(log2 > 0, na.rm = TRUE),
  under = sum(log2 < 0, na.rm = TRUE),
  
 ) %>%
 filter(
  n > 100
 )
row.names(class.table.root) <- class.table.root$tax_class
others.class.root <- data.frame("Others", (23274 - sum(class.table.root$n)), (23042 - sum(class.table.root$over)), (232 - sum(class.table.root$under)))
colnames(others.class.root) <- colnames(class.table.root)
row.names(others.class.root) <- "Others"
class.table.root = rbind(class.table.root, others.class.root) 
colSums(class.table.root[,2:4]) #Should be 23274 23042 232

#Get a column for all transcripts
annot.fun <- annot[grep(".mycota", annot$tax_phylum),]
annot.fun.root <- annot.fun[annot.fun$gene_id %in% row.names(MT.root),]
MT.root.fun <- MT.root[row.names(MT.root) %in% annot.fun$gene_id,]
annot.fun.root <- annot.fun.root[order(annot.fun.root$gene_id),]
MT.root.fun <- MT.root.fun[order(row.names(MT.root.fun)),]
sum(row.names(MT.root.fun) == annot.fun.root$gene_id)
MT.root.fun.tax <- data.frame(cbind(MT.root.fun, annot.fun.root$tax_class))

all.column.root = MT.root.fun.tax %>%
 group_by(V12) %>%
 summarise(
  count = n(),
  
  
 )
all.column.root <- all.column.root[all.column.root$V12 %in% class.table.root$tax_class,]
row.names(all.column.root) <- all.column.root$V12
others.all.root <- data.frame("Others", (200233 - sum(all.column.root$count)))
colnames(others.all.root) <- colnames(all.column.root)
row.names(others.all.root) <- "Others"
all.column.root <- rbind(all.column.root, others.all.root) 
sum(all.column.root$count)

all.column.root$V12 == class.table.root$tax_class
class.table.root <- cbind(class.table.root,all.column.root$count)
colnames(class.table.root) <- c("Class", "n", "More", "Less", "All")

#Prepare for ggplot
class.table.root.long <- gather(class.table.root,subset,counts,3:5) #transform in long format for ggplot
palette(c(brewer.pal(n = 9, name = "Set1"),"lightgrey", "black", "darkred", "darkblue", "darkgreen", "purple4", "darkgrey", "white"))
stack.root.class <- ggplot(class.table.root.long, aes(fill = Class, y = counts, x = subset)) + 
  geom_bar( stat = "identity", position = "fill") +
  ylab("Fraction of reads") + 
  scale_fill_manual(values = palette(), guide = guide_legend(label.theme = element_text(face = "italic", size = 8))) +
  theme_bw() +
  scale_y_continuous( expand = c(0,0)) +
  scale_x_discrete(name = "Subset") +
  theme(axis.text.x = element_text(size = 8, angle = 45, hjust = 1))
stack.root.class
```

```{r stack barchart COG root}
#COG for FUNGI
#Create data frame
DEfound.root.fun <- DEfound.root[grep(".mycota", DEfound.root$tax_phylum),]
COG.root <- DEfound.root.fun %>%
 group_by(cog_category) %>%
 summarise(
  n = n(),
  over = sum(log2 > 0, na.rm = TRUE),
  under = sum(log2 < 0, na.rm = TRUE),
  
 ) %>%
 filter(
  n > 100
 )
COG.root <- COG.root[-c(4,9,13,14,16,17),] #Remove some categories mixed and NULL... NULL: 14041 13901 140; 
row.names(COG.root) <- COG.root$cog_category
others.cog.root <- data.frame("Others", (23274 - 14041 - sum(COG.root$n)), (23042 - 13901 - sum(COG.root$over)), (232 - 140 - um(COG.root$under))) #Create other bu omit NULL
colnames(others.cog.root) <- colnames(COG.root)
row.names(others.cog.root) = "Others"
COG.root <- rbind(COG.root, others.cog.root) 
colSums(COG.root[,2:4])

#Get a column for all fungal transcripts
annot.fun <- annot[grep(".mycota", annot$tax_phylum),]
annot.fun.root <- annot.fun[annot.fun$gene_id %in% row.names(MT.root),]
MT.root.fun <- MT.root[row.names(MT.root) %in% annot.fun$gene_id,]
annot.fun.root <- annot.fun.root[order(annot.fun.root$gene_id),]
MT.root.fun <- MT.root.fun[order(row.names(MT.root.fun)),]
sum(row.names(MT.root.fun) == annot.fun.root$gene_id)
MT.root.fun.cog <- data.frame(cbind(MT.root.fun, annot.fun.root$cog_category))

COG.root.all = MT.root.fun.cog %>%
 group_by(V12) %>%
 summarise(
  count = n(),
  
  
 )
COG.root.all <- COG.root.all[COG.root.all$V12 %in% COG.root$cog_category,]
row.names(COG.root.all) <- COG.root.all$V12
others.cog.root.all <- data.frame("Others", (200233 - 134958 - sum(COG.root.all$count))) #Omit NULL
colnames(others.cog.root.all) <- colnames(COG.root.all)
row.names(others.cog.root.all) <- "Others"
COG.root.all <- rbind(COG.root.all, others.cog.root.all) 
sum(COG.root.all$count)

COG.root.all$V12 == COG.root$cog_category
COG.root <- cbind(COG.root,COG.root.all$count)
colnames(COG.root) <- c("COG_category", "n", "More", "Less", "All")

#Prepare for ggplot
COG.root.long <- gather(COG.root,subset,counts,3:5) #transform in long format for ggplot
palette(c(brewer.pal(n = 9, name = "Set1"),"lightgrey", "black", "darkred", "darkblue", "darkgreen", "purple4", "brown3", "cyan", "pink"))
stack.COG.root <- ggplot(COG.root.long, aes(fill = COG_category, y = counts, x = subset)) + 
  geom_bar( stat = "identity", position = "fill") +
  ylab("Fraction of reads") + 
  scale_fill_manual(values = palette(), guide = guide_legend(label.theme = element_text(face = "plain", size = 8))) +
  theme_bw() +
  scale_y_continuous( expand = c(0,0)) +
  scale_x_discrete(name = "Subset") +
  theme(axis.text.x = element_text(size = 8, angle = 45, hjust = 1))
stack.COG.root
```

```{r stack barchart COG Rhizo}
#COG for BACTERIA
#Create data frame
DEfound.rhizo.bact <- DEfound.rhizo[DEfound.rhizo$tax_kingdom == "Bacteria",]
COG.rhizo <- DEfound.rhizo.bact %>%
 group_by(cog_category) %>%
 summarise(
  n = n(),
  over = sum(log2 > 0, na.rm = TRUE),
  under = sum(log2 < 0, na.rm = TRUE),
  
 ) %>%
 filter(
  n > 100
 )
COG.rhizo <- COG.rhizo[-c(4,5,13),] #Remove NULL: 6865 3759 3106; Cell motility: 199 1 198; Coenzyme: 134 91 43; to have matching legends with ROOT
row.names(COG.rhizo) <- COG.rhizo$cog_category
others.COG.rhizo <- data.frame("Others", (14178 - 6865 - sum(COG.rhizo$n)), (7938 - 3759 - sum(COG.rhizo$over)), (6240 - 3106 - sum(COG.rhizo$under))) #Omit NULL from Others count
colnames(others.COG.rhizo) <- colnames(COG.rhizo)
row.names(others.COG.rhizo) <- "Others"
COG.rhizo <- rbind(COG.rhizo, others.COG.rhizo) 
colSums(COG.rhizo[,2:4]) #7313 4179 3134

#Get a column for all transcripts
annot.bact <- annot[annot$tax_kingdom == "Bacteria",]
annot.bact.rhizo <- annot.bact[annot.bact$gene_id %in% row.names(MT.rhizo),]
MT.rhizo.bact <- MT.rhizo[row.names(MT.rhizo) %in% annot.bact$gene_id,]
annot.bact.rhizo <- annot.bact.rhizo[order(annot.bact.rhizo$gene_id),]
MT.rhizo.bact <- MT.rhizo.bact[order(row.names(MT.rhizo.bact)),]
sum(row.names(MT.rhizo.bact) == annot.bact.rhizo$gene_id)
MT.rhizo.bact.cog <- data.frame(cbind(MT.rhizo.bact, annot.bact.rhizo$cog_category))

COG.rhizo.all = MT.rhizo.bact.cog %>%
 group_by(V13) %>%
 summarise(
  count = n(),
  
  
 )
COG.rhizo.all <- COG.rhizo.all[COG.rhizo.all$V13 %in% COG.rhizo$cog_category,]
row.names(COG.rhizo.all) <- COG.rhizo.all$V13
others.COG.rhizo.all <- data.frame("Others", (430984 - 206920 - sum(COG.rhizo.all$count))) #Omit NULL
colnames(others.COG.rhizo.all) <- colnames(COG.rhizo.all)
row.names(others.COG.rhizo.all) <- "Others"
COG.rhizo.all <- rbind(COG.rhizo.all, others.COG.rhizo.all) 
sum(COG.rhizo.all$count)

COG.rhizo.all$V13 == COG.rhizo$cog_category
COG.rhizo <- cbind(COG.rhizo,COG.rhizo.all$count)
colnames(COG.rhizo) <- c("COG_category", "n", "More", "Less", "All")

#Prepare for ggplot
rhizo.COG <- gather(COG.rhizo,subset,counts,3:5) #transform in long format for ggplot
palette(c(brewer.pal(n = 9, name = "Set1"),"lightgrey", "black", "darkred", "darkblue", "darkgreen", "purple4", "brown3", "cyan"))
stack.COG.rhizo <- ggplot(rhizo.COG, aes(fill = COG_category, y = counts, x = subset)) + 
  geom_bar( stat = "identity", position = "fill") +
  ylab("Fraction of reads") + 
  scale_fill_manual(values = palette(), guide = guide_legend(label.theme = element_text(face = "plain", size = 8))) +
  theme_bw() +
  scale_y_continuous( expand = c(0,0)) +
  scale_x_discrete(name = "Subset") +
  theme(axis.text.x = element_text(size = 8, angle = 45, hjust = 1))
stack.COG.rhizo  
```

```{r top 50 DE genes roots and rhizo}
###ROOTS
#Sort based on P-value of EE
DEfound.root.50 <- DEfound.root[DEfound.root$PPEE == 0,] #lots of Pvalue = 0... 1491. Then take most abundant.
DEfound.root.50 <- DEfound.root.50[order(DEfound.root.50$mean, decreasing = TRUE),]
DEfound.root.50.defined <- DEfound.root.50[DEfound.root.50$cog_function != "NULL",] #Remove undefined
DEfound.root.50.defined <- DEfound.root.50.defined[1:50, c(1:4, 24,25,34,37)] #Keep omly relevant columns
#Export
write.table(DEfound.root.50.defined, "../output/Table1.txt", eol = "\n", sep = "\t")

###RHIZO
#Sort based on P-value of EE
DEfound.rhizo.50 <- DEfound.rhizo[order(DEfound.rhizo$PPEE),]
DEfound.rhizo.50.defined <- DEfound.rhizo.50[DEfound.rhizo.50$cog_function != "NULL",] #Remove undefined
DEfound.rhizo.50.defined <- DEfound.rhizo.50.defined[1:100, c(1:4, 24,25,34,37)] #Keep only relevant columns, keep 100, because lot of irrelevant orgs; need to manually remove
#Export
write.table(DEfound.rhizo.50.defined, "../output/Table2.txt", eol = "\n", sep = "\t")

```


```{r anova taxa Rhizo}
#Prokaryotes at phylum
#Get table with all reps
MT.rhizo.bact.tax.found <- MT.rhizo.bact.tax[row.names(MT.rhizo.bact.tax) %in% row.names(DEfound.rhizo.bact),] #Get only DEfound
DEfound.rhizo.bact.s <- DEfound.rhizo.bact[order(row.names(DEfound.rhizo.bact)),] #Sort
sum(row.names(DEfound.rhizo.bact.s) == row.names(MT.rhizo.bact.tax.found)) #14178
MT.rhizo.bact.tax.found <- data.frame(cbind(MT.rhizo.bact.tax.found, DEfound.rhizo.bact.s$log2))#Add column for log2
MT.rhizo.bact.tax.found.over <- MT.rhizo.bact.tax.found[DEfound.rhizo.bact.s$log2 > 0,]
MT.rhizo.bact.tax.found.under <- MT.rhizo.bact.tax.found[DEfound.rhizo.bact.s$log2 < 0,]
MT.rhizo.bact.tax.found.over <- data.frame(apply(MT.rhizo.bact.tax.found.over[,1:12], 2,      # Specify own function within apply
         function(x) as.numeric(as.character(x))), MT.rhizo.bact.tax.found.over[,13:14])
MT.rhizo.bact.tax.found.under <- data.frame(apply(MT.rhizo.bact.tax.found.under[,1:12], 2,      # Specify own function within apply
         function(x) as.numeric(as.character(x))), MT.rhizo.bact.tax.found.under[,13:14])
MT.rhizo.bact.tax <- data.frame(apply(MT.rhizo.bact.tax[,1:12], 2,      # Specify own function within apply
         function(x) as.numeric(as.character(x))), MT.rhizo.bact.tax[,13])



#DEFound, only group_by phylum, keep all samples
summary.rhizo.bact.over <- MT.rhizo.bact.tax.found.over %>%
 group_by(V13) %>%
 summarise(
  B1P2S.over = sum(B1P2S),
  B1P3S.over = sum(B1P3S),
  B2P1S.over = sum(B2P1S),
  B2P2S.over = sum(B2P2S),
  B3P2S.over = sum(B3P2S),
  B3P8S.over = sum(B3P8S),
  B4P5S.over = sum(B4P5S),
  B4P7S.over = sum(B4P7S),
  B5P2S.over = sum(B5P2S),
  B5P3S.over = sum(B5P3S),
  B6P2S.over = sum(B6P2S),
  B6P6S.over = sum(B6P6S),
  
 )
summary.rhizo.bact.under <- MT.rhizo.bact.tax.found.under %>%
 group_by(V13) %>%
 summarise(
  B1P2S.under = sum(B1P2S),
  B1P3S.under = sum(B1P3S),
  B2P1S.under = sum(B2P1S),
  B2P2S.under = sum(B2P2S),
  B3P2S.under = sum(B3P2S),
  B3P8S.under = sum(B3P8S),
  B4P5S.under = sum(B4P5S),
  B4P7S.under = sum(B4P7S),
  B5P2S.under = sum(B5P2S),
  B5P3S.under = sum(B5P3S),
  B6P2S.under = sum(B6P2S),
  B6P6S.under = sum(B6P6S),
  
 )

summary.rhizo.bact.all <- MT.rhizo.bact.tax %>%
 group_by(MT.rhizo.bact.tax...13.) %>%
 summarise(
  B1P2S.all = sum(B1P2S),
  B1P3S.all = sum(B1P3S),
  B2P1S.all = sum(B2P1S),
  B2P2S.all = sum(B2P2S),
  B3P2S.all = sum(B3P2S),
  B3P8S.all = sum(B3P8S),
  B4P5S.all = sum(B4P5S),
  B4P7S.all = sum(B4P7S),
  B5P2S.all = sum(B5P2S),
  B5P3S.all = sum(B5P3S),
  B6P2S.all = sum(B6P2S),
  B6P6S.all = sum(B6P6S),
  
 )

#Put everything in relative
#All
summary.rhizo.bact.all$Phylum <- summary.rhizo.bact.all$MT.rhizo.bact.tax...13.
summary.rhizo.bact.all <- summary.rhizo.bact.all[,-1]
summary.rhizo.bact.all.rel <- data.frame(t(apply(summary.rhizo.bact.all[,1:12], 1, "/", colSums(summary.rhizo.bact.all[,1:12]))))
colSums(summary.rhizo.bact.all.rel)
summary.rhizo.bact.all.rel$Phylum <- summary.rhizo.bact.all$Phylum
#Over
summary.rhizo.bact.over$Phylum <- summary.rhizo.bact.over$V13
summary.rhizo.bact.over <- summary.rhizo.bact.over[,-1]
summary.rhizo.bact.over.rel <- data.frame(t(apply(summary.rhizo.bact.over[,1:12], 1, "/", colSums(summary.rhizo.bact.over[,1:12]))))
colSums(summary.rhizo.bact.over.rel)
summary.rhizo.bact.over.rel$Phylum <- summary.rhizo.bact.over$Phylum
#under
summary.rhizo.bact.under$Phylum <- summary.rhizo.bact.under$V13
summary.rhizo.bact.under <- summary.rhizo.bact.under[,-1]
summary.rhizo.bact.under.rel <- data.frame(t(apply(summary.rhizo.bact.under[,1:12], 1, "/", colSums(summary.rhizo.bact.under[,1:12]))))
colSums(summary.rhizo.bact.under.rel)
summary.rhizo.bact.under.rel$Phylum <- summary.rhizo.bact.under$Phylum

#ALL vs. OVER
#Acidobacteria
"All:"; summary.rhizo.bact.all.rel[2,13]; mean(as.numeric(summary.rhizo.bact.all.rel[2,1:12]))
"Over:"; summary.rhizo.bact.over.rel[2,13]; mean(as.numeric(summary.rhizo.bact.over.rel[2,1:12]))
t.test(t(summary.rhizo.bact.all.rel[2,1:12]), t(summary.rhizo.bact.over.rel[2,1:12]), paired = TRUE)
#Actinobacteria
"All:"; summary.rhizo.bact.all.rel[3,13]; mean(as.numeric(summary.rhizo.bact.all.rel[3,1:12]))
"Over:"; summary.rhizo.bact.over.rel[3,13]; mean(as.numeric(summary.rhizo.bact.over.rel[3,1:12]))
t.test(t(summary.rhizo.bact.all.rel[3,1:12]), t(summary.rhizo.bact.over.rel[3,1:12]), paired = TRUE)
#Bacteroidetes
"All:"; summary.rhizo.bact.all.rel[6,13]; mean(as.numeric(summary.rhizo.bact.all.rel[6,1:12]))
"Over:"; summary.rhizo.bact.over.rel[5,13]; mean(as.numeric(summary.rhizo.bact.over.rel[5,1:12]))
t.test(t(summary.rhizo.bact.all.rel[6,1:12]), t(summary.rhizo.bact.over.rel[5,1:12]), paired = TRUE)
#Candidatus Rokubacteria
"All:"; summary.rhizo.bact.all.rel[91,13]; mean(as.numeric(summary.rhizo.bact.all.rel[91,1:12]))
"Over:"; summary.rhizo.bact.over.rel[20,13]; mean(as.numeric(summary.rhizo.bact.over.rel[20,1:12]))
t.test(t(summary.rhizo.bact.all.rel[91,1:12]), t(summary.rhizo.bact.over.rel[20,1:12]), paired = TRUE)
#Chloroflexi
"All:"; summary.rhizo.bact.all.rel[115,13]; mean(as.numeric(summary.rhizo.bact.all.rel[115,1:12]))
"Over:"; summary.rhizo.bact.over.rel[24,13]; mean(as.numeric(summary.rhizo.bact.over.rel[24,1:12]))
t.test(t(summary.rhizo.bact.all.rel[115,1:12]), t(summary.rhizo.bact.over.rel[24,1:12]), paired = TRUE)
#Cyanobacteria
"All:"; summary.rhizo.bact.all.rel[118,13]; mean(as.numeric(summary.rhizo.bact.all.rel[118,1:12]))
"Over:"; summary.rhizo.bact.over.rel[25,13]; mean(as.numeric(summary.rhizo.bact.over.rel[25,1:12]))
t.test(t(summary.rhizo.bact.all.rel[118,1:12]), t(summary.rhizo.bact.over.rel[25,1:12]), paired = TRUE)
#Gemmatimonadetes
"All:"; summary.rhizo.bact.all.rel[126,13]; mean(as.numeric(summary.rhizo.bact.all.rel[126,1:12]))
"Over:"; summary.rhizo.bact.over.rel[29,13]; mean(as.numeric(summary.rhizo.bact.over.rel[29,1:12]))
t.test(t(summary.rhizo.bact.all.rel[126,1:12]), t(summary.rhizo.bact.over.rel[29,1:12]), paired = TRUE)
#Nitrospira
"All:"; summary.rhizo.bact.all.rel[131,13]; mean(as.numeric(summary.rhizo.bact.all.rel[131,1:12]))
"Over:"; summary.rhizo.bact.over.rel[33,13]; mean(as.numeric(summary.rhizo.bact.over.rel[33,1:12]))
t.test(t(summary.rhizo.bact.all.rel[131,1:12]), t(summary.rhizo.bact.over.rel[33,1:12]), paired = TRUE)
#Planctomycetes
"All:"; summary.rhizo.bact.all.rel[133,13]; mean(as.numeric(summary.rhizo.bact.all.rel[133,1:12]))
"Over:"; summary.rhizo.bact.over.rel[35,13]; mean(as.numeric(summary.rhizo.bact.over.rel[35,1:12]))
t.test(t(summary.rhizo.bact.all.rel[133,1:12]), t(summary.rhizo.bact.over.rel[35,1:12]), paired = TRUE)
#Proteobacteria
"All:"; summary.rhizo.bact.all.rel[134,13]; mean(as.numeric(summary.rhizo.bact.all.rel[134,1:12]))
"Over:"; summary.rhizo.bact.over.rel[36,13]; mean(as.numeric(summary.rhizo.bact.over.rel[36,1:12]))
t.test(t(summary.rhizo.bact.all.rel[134,1:12]), t(summary.rhizo.bact.over.rel[36,1:12]), paired = TRUE)
#Verrucomicrobia
"All:"; summary.rhizo.bact.all.rel[141,13]; mean(as.numeric(summary.rhizo.bact.all.rel[141,1:12]))
"Over:"; summary.rhizo.bact.over.rel[39,13]; mean(as.numeric(summary.rhizo.bact.over.rel[39,1:12]))
t.test(t(summary.rhizo.bact.all.rel[141,1:12]), t(summary.rhizo.bact.over.rel[39,1:12]), paired = TRUE)

#ALL vs. UNDER
#Acidobacteria
"All:"; summary.rhizo.bact.all.rel[2,13]; mean(as.numeric(summary.rhizo.bact.all.rel[2,1:12]))
"under:"; summary.rhizo.bact.under.rel[2,13]; mean(as.numeric(summary.rhizo.bact.under.rel[2,1:12]))
t.test(t(summary.rhizo.bact.all.rel[2,1:12]), t(summary.rhizo.bact.under.rel[2,1:12]), paired = TRUE)
#Actinobacteria
"All:"; summary.rhizo.bact.all.rel[3,13]; mean(as.numeric(summary.rhizo.bact.all.rel[3,1:12]))
"under:"; summary.rhizo.bact.under.rel[3,13]; mean(as.numeric(summary.rhizo.bact.under.rel[3,1:12]))
t.test(t(summary.rhizo.bact.all.rel[3,1:12]), t(summary.rhizo.bact.under.rel[3,1:12]), paired = TRUE)
#Bacteroidetes
"All:"; summary.rhizo.bact.all.rel[6,13]; mean(as.numeric(summary.rhizo.bact.all.rel[6,1:12]))
"under:"; summary.rhizo.bact.under.rel[6,13]; mean(as.numeric(summary.rhizo.bact.under.rel[6,1:12]))
t.test(t(summary.rhizo.bact.all.rel[6,1:12]), t(summary.rhizo.bact.under.rel[6,1:12]), paired = TRUE)
#Candidatus Rokubacteria
"All:"; summary.rhizo.bact.all.rel[91,13]; mean(as.numeric(summary.rhizo.bact.all.rel[91,1:12]))
"under:"; summary.rhizo.bact.under.rel[35,13]; mean(as.numeric(summary.rhizo.bact.under.rel[35,1:12]))
t.test(t(summary.rhizo.bact.all.rel[91,1:12]), t(summary.rhizo.bact.under.rel[35,1:12]), paired = TRUE)
#Chloroflexi
"All:"; summary.rhizo.bact.all.rel[115,13]; mean(as.numeric(summary.rhizo.bact.all.rel[115,1:12]))
"under:"; summary.rhizo.bact.under.rel[46,13]; mean(as.numeric(summary.rhizo.bact.under.rel[46,1:12]))
t.test(t(summary.rhizo.bact.all.rel[115,1:12]), t(summary.rhizo.bact.under.rel[46,1:12]), paired = TRUE)
#Cyanobacteria
"All:"; summary.rhizo.bact.all.rel[118,13]; mean(as.numeric(summary.rhizo.bact.all.rel[118,1:12]))
"under:"; summary.rhizo.bact.under.rel[47,13]; mean(as.numeric(summary.rhizo.bact.under.rel[47,1:12]))
t.test(t(summary.rhizo.bact.all.rel[118,1:12]), t(summary.rhizo.bact.under.rel[47,1:12]), paired = TRUE)
#Gemmatimonadetes
"All:"; summary.rhizo.bact.all.rel[126,13]; mean(as.numeric(summary.rhizo.bact.all.rel[126,1:12]))
"under:"; summary.rhizo.bact.under.rel[52,13]; mean(as.numeric(summary.rhizo.bact.under.rel[52,1:12]))
t.test(t(summary.rhizo.bact.all.rel[126,1:12]), t(summary.rhizo.bact.under.rel[52,1:12]), paired = TRUE)
#Nitrospira
"All:"; summary.rhizo.bact.all.rel[131,13]; mean(as.numeric(summary.rhizo.bact.all.rel[131,1:12]))
"under:"; summary.rhizo.bact.under.rel[56,13]; mean(as.numeric(summary.rhizo.bact.under.rel[56,1:12]))
t.test(t(summary.rhizo.bact.all.rel[131,1:12]), t(summary.rhizo.bact.under.rel[56,1:12]), paired = TRUE)
#Planctomycetes
"All:"; summary.rhizo.bact.all.rel[133,13]; mean(as.numeric(summary.rhizo.bact.all.rel[133,1:12]))
"under:"; summary.rhizo.bact.under.rel[58,13]; mean(as.numeric(summary.rhizo.bact.under.rel[58,1:12]))
t.test(t(summary.rhizo.bact.all.rel[133,1:12]), t(summary.rhizo.bact.under.rel[58,1:12]), paired = TRUE)
#Proteobacteria
"All:"; summary.rhizo.bact.all.rel[134,13]; mean(as.numeric(summary.rhizo.bact.all.rel[134,1:12]))
"under:"; summary.rhizo.bact.under.rel[59,13]; mean(as.numeric(summary.rhizo.bact.under.rel[59,1:12]))
t.test(t(summary.rhizo.bact.all.rel[134,1:12]), t(summary.rhizo.bact.under.rel[59,1:12]), paired = TRUE)
#Verrucomicrobia
"All:"; summary.rhizo.bact.all.rel[141,13]; mean(as.numeric(summary.rhizo.bact.all.rel[141,1:12]))
"under:"; summary.rhizo.bact.under.rel[65,13]; mean(as.numeric(summary.rhizo.bact.under.rel[65,1:12]))
t.test(t(summary.rhizo.bact.all.rel[141,1:12]), t(summary.rhizo.bact.under.rel[65,1:12]), paired = TRUE)

```

```{r anova COG Rhizo}
#Prokaryotes at phylum
#Get table with all reps
MT.rhizo.bact.cog.found <- MT.rhizo.bact.cog[row.names(MT.rhizo.bact.cog) %in% row.names(DEfound.rhizo.bact),] #Get only DEfound
DEfound.rhizo.bact.s <- DEfound.rhizo.bact[order(row.names(DEfound.rhizo.bact)),] #Sort
sum(row.names(DEfound.rhizo.bact.s) == row.names(MT.rhizo.bact.cog.found)) #14178
MT.rhizo.bact.cog.found <- data.frame(cbind(MT.rhizo.bact.cog.found, DEfound.rhizo.bact.s$log2))#Add column for log2
MT.rhizo.bact.cog.found.over <- MT.rhizo.bact.cog.found[DEfound.rhizo.bact.s$log2 > 0,]
MT.rhizo.bact.cog.found.under <- MT.rhizo.bact.cog.found[DEfound.rhizo.bact.s$log2 < 0,]
MT.rhizo.bact.cog.found.over <- data.frame(apply(MT.rhizo.bact.cog.found.over[,1:12], 2,  # Specify own function within apply
         function(x) as.numeric(as.character(x))), MT.rhizo.bact.cog.found.over[,13:14])
MT.rhizo.bact.cog.found.under <- data.frame(apply(MT.rhizo.bact.cog.found.under[,1:12], 2,  # Specify own function within apply
         function(x) as.numeric(as.character(x))), MT.rhizo.bact.cog.found.under[,13:14])
MT.rhizo.bact.cog <- data.frame(apply(MT.rhizo.bact.cog[,1:12], 2, # Specify own function within apply
         function(x) as.numeric(as.character(x))), MT.rhizo.bact.cog[,13])



#DEFound, only group_by phylum, keep all samples
summary.rhizo.cog.over <- MT.rhizo.bact.cog.found.over %>%
 group_by(V13) %>%
 summarise(
  B1P2S.over = sum(B1P2S),
  B1P3S.over = sum(B1P3S),
  B2P1S.over = sum(B2P1S),
  B2P2S.over = sum(B2P2S),
  B3P2S.over = sum(B3P2S),
  B3P8S.over = sum(B3P8S),
  B4P5S.over = sum(B4P5S),
  B4P7S.over = sum(B4P7S),
  B5P2S.over = sum(B5P2S),
  B5P3S.over = sum(B5P3S),
  B6P2S.over = sum(B6P2S),
  B6P6S.over = sum(B6P6S),
  
 )
summary.rhizo.cog.under <- MT.rhizo.bact.cog.found.under %>%
 group_by(V13) %>%
 summarise(
  B1P2S.under = sum(B1P2S),
  B1P3S.under = sum(B1P3S),
  B2P1S.under = sum(B2P1S),
  B2P2S.under = sum(B2P2S),
  B3P2S.under = sum(B3P2S),
  B3P8S.under = sum(B3P8S),
  B4P5S.under = sum(B4P5S),
  B4P7S.under = sum(B4P7S),
  B5P2S.under = sum(B5P2S),
  B5P3S.under = sum(B5P3S),
  B6P2S.under = sum(B6P2S),
  B6P6S.under = sum(B6P6S),
  
 )

summary.rhizo.cog.all <- MT.rhizo.bact.cog %>%
 group_by(MT.rhizo.bact.cog...13.) %>%
 summarise(
  B1P2S.all = sum(B1P2S),
  B1P3S.all = sum(B1P3S),
  B2P1S.all = sum(B2P1S),
  B2P2S.all = sum(B2P2S),
  B3P2S.all = sum(B3P2S),
  B3P8S.all = sum(B3P8S),
  B4P5S.all = sum(B4P5S),
  B4P7S.all = sum(B4P7S),
  B5P2S.all = sum(B5P2S),
  B5P3S.all = sum(B5P3S),
  B6P2S.all = sum(B6P2S),
  B6P6S.all = sum(B6P6S),
  
 )

#Put everything in relative
#All
summary.rhizo.cog.all$COG_cat <- summary.rhizo.cog.all$MT.rhizo.bact.cog...13.
summary.rhizo.cog.all <- summary.rhizo.cog.all[,-1]
summary.rhizo.cog.all.rel <- data.frame(t(apply(summary.rhizo.cog.all[,1:12], 1, "/", colSums(summary.rhizo.cog.all[,1:12]))))
colSums(summary.rhizo.cog.all.rel)
summary.rhizo.cog.all.rel$COG_cat <- summary.rhizo.cog.all$COG_cat
#Over
summary.rhizo.cog.over$COG_cat <- summary.rhizo.cog.over$V13
summary.rhizo.cog.over <- summary.rhizo.cog.over[,-1]
summary.rhizo.cog.over.rel <- data.frame(t(apply(summary.rhizo.cog.over[,1:12], 1, "/", colSums(summary.rhizo.cog.over[,1:12]))))
colSums(summary.rhizo.cog.over.rel)
summary.rhizo.cog.over.rel$COG_cat <- summary.rhizo.cog.over$COG_cat
#under
summary.rhizo.cog.under$COG_cat <- summary.rhizo.cog.under$V13
summary.rhizo.cog.under <- summary.rhizo.cog.under[,-1]
summary.rhizo.cog.under.rel <- data.frame(t(apply(summary.rhizo.cog.under[,1:12], 1, "/", colSums(summary.rhizo.cog.under[,1:12]))))
colSums(summary.rhizo.cog.under.rel)
summary.rhizo.cog.under.rel$COG_cat <- summary.rhizo.cog.under$COG_cat

#ALL vs. OVER
#AA trans and metab
"All:"; summary.rhizo.cog.all.rel[6,13]; mean(as.numeric(summary.rhizo.cog.all.rel[6,1:12]))
"Over:"; summary.rhizo.cog.over.rel[3,13]; mean(as.numeric(summary.rhizo.cog.over.rel[3,1:12]))
t.test(t(summary.rhizo.cog.all.rel[6,1:12]), t(summary.rhizo.cog.over.rel[3,1:12]), paired = TRUE)
#Carb trans and metabolism
"All:"; summary.rhizo.cog.all.rel[17,13]; mean(as.numeric(summary.rhizo.cog.all.rel[17,1:12]))
"Over:"; summary.rhizo.cog.over.rel[9,13]; mean(as.numeric(summary.rhizo.cog.over.rel[9,1:12]))
t.test(t(summary.rhizo.cog.all.rel[17,1:12]), t(summary.rhizo.cog.over.rel[9,1:12]), paired = TRUE)
#Cell envelope
"All:"; summary.rhizo.cog.all.rel[28,13]; mean(as.numeric(summary.rhizo.cog.all.rel[28,1:12]))
"Over:"; summary.rhizo.cog.over.rel[16,13]; mean(as.numeric(summary.rhizo.cog.over.rel[16,1:12]))
t.test(t(summary.rhizo.cog.all.rel[28,1:12]), t(summary.rhizo.cog.over.rel[16,1:12]), paired = TRUE)
#Cell motility and secretion
"All:"; summary.rhizo.cog.all.rel[36,13]; mean(as.numeric(summary.rhizo.cog.all.rel[36,1:12]))
"Over:"; summary.rhizo.cog.over.rel[21,13]; mean(as.numeric(summary.rhizo.cog.over.rel[21,1:12]))
t.test(t(summary.rhizo.cog.all.rel[36,1:12]), t(summary.rhizo.cog.over.rel[21,1:12]), paired = TRUE)
#Coenzyme metabolism
"All:"; summary.rhizo.cog.all.rel[45,13]; mean(as.numeric(summary.rhizo.cog.all.rel[45,1:12]))
"Over:"; summary.rhizo.cog.over.rel[25,13]; mean(as.numeric(summary.rhizo.cog.over.rel[25,1:12]))
t.test(t(summary.rhizo.cog.all.rel[45,1:12]), t(summary.rhizo.cog.over.rel[25,1:12]), paired = TRUE)
#DNA replication
"All:"; summary.rhizo.cog.all.rel[58,13]; mean(as.numeric(summary.rhizo.cog.all.rel[58,1:12]))
"Over:"; summary.rhizo.cog.over.rel[32,13]; mean(as.numeric(summary.rhizo.cog.over.rel[32,1:12]))
t.test(t(summary.rhizo.cog.all.rel[58,1:12]), t(summary.rhizo.cog.over.rel[32,1:12]), paired = TRUE)
#Energy production and conversion
"All:"; summary.rhizo.cog.all.rel[64,13]; mean(as.numeric(summary.rhizo.cog.all.rel[64,1:12]))
"Over:"; summary.rhizo.cog.over.rel[35,13]; mean(as.numeric(summary.rhizo.cog.over.rel[35,1:12]))
t.test(t(summary.rhizo.cog.all.rel[64,1:12]), t(summary.rhizo.cog.over.rel[35,1:12]), paired = TRUE)
#Inorganic ion
"All:"; summary.rhizo.cog.all.rel[77,13]; mean(as.numeric(summary.rhizo.cog.all.rel[77,1:12]))
"Over:"; summary.rhizo.cog.over.rel[42,13]; mean(as.numeric(summary.rhizo.cog.over.rel[42,1:12]))
t.test(t(summary.rhizo.cog.all.rel[77,1:12]), t(summary.rhizo.cog.over.rel[42,1:12]), paired = TRUE)
#Intracellular trafficking
"All:"; summary.rhizo.cog.all.rel[83,13]; mean(as.numeric(summary.rhizo.cog.all.rel[83,1:12]))
"Over:"; summary.rhizo.cog.over.rel[45,13]; mean(as.numeric(summary.rhizo.cog.over.rel[45,1:12]))
t.test(t(summary.rhizo.cog.all.rel[83,1:12]), t(summary.rhizo.cog.over.rel[45,1:12]), paired = TRUE)
#Lipid
"All:"; summary.rhizo.cog.all.rel[89,13]; mean(as.numeric(summary.rhizo.cog.all.rel[89,1:12]))
"Over:"; summary.rhizo.cog.over.rel[48,13]; mean(as.numeric(summary.rhizo.cog.over.rel[48,1:12]))
t.test(t(summary.rhizo.cog.all.rel[89,1:12]), t(summary.rhizo.cog.over.rel[48,1:12]), paired = TRUE)
#Posttranslational
"All:"; summary.rhizo.cog.all.rel[102,13]; mean(as.numeric(summary.rhizo.cog.all.rel[102,1:12]))
"Over:"; summary.rhizo.cog.over.rel[55,13]; mean(as.numeric(summary.rhizo.cog.over.rel[55,1:12]))
t.test(t(summary.rhizo.cog.all.rel[102,1:12]), t(summary.rhizo.cog.over.rel[55,1:12]), paired = TRUE)
#Signal
"All:"; summary.rhizo.cog.all.rel[125,13]; mean(as.numeric(summary.rhizo.cog.all.rel[125,1:12]))
"Over:"; summary.rhizo.cog.over.rel[64,13]; mean(as.numeric(summary.rhizo.cog.over.rel[64,1:12]))
t.test(t(summary.rhizo.cog.all.rel[125,1:12]), t(summary.rhizo.cog.over.rel[64,1:12]), paired = TRUE)
#Transcription
"All:"; summary.rhizo.cog.all.rel[129,13]; mean(as.numeric(summary.rhizo.cog.all.rel[129,1:12]))
"Over:"; summary.rhizo.cog.over.rel[67,13]; mean(as.numeric(summary.rhizo.cog.over.rel[67,1:12]))
t.test(t(summary.rhizo.cog.all.rel[129,1:12]), t(summary.rhizo.cog.over.rel[67,1:12]), paired = TRUE)
#Translation
"All:"; summary.rhizo.cog.all.rel[135,13]; mean(as.numeric(summary.rhizo.cog.all.rel[135,1:12]))
"Over:"; summary.rhizo.cog.over.rel[72,13]; mean(as.numeric(summary.rhizo.cog.over.rel[72,1:12]))
t.test(t(summary.rhizo.cog.all.rel[135,1:12]), t(summary.rhizo.cog.over.rel[72,1:12]), paired = TRUE)

#ALL vs. UNDER
#AA trans and metab
"All:"; summary.rhizo.cog.all.rel[6,13]; mean(as.numeric(summary.rhizo.cog.all.rel[6,1:12]))
"Under:"; summary.rhizo.cog.under.rel[4,13]; mean(as.numeric(summary.rhizo.cog.under.rel[4,1:12]))
t.test(t(summary.rhizo.cog.all.rel[6,1:12]), t(summary.rhizo.cog.under.rel[4,1:12]), paired = TRUE)
#Carb trans and metabolism
"All:"; summary.rhizo.cog.all.rel[17,13]; mean(as.numeric(summary.rhizo.cog.all.rel[17,1:12]))
"Under:"; summary.rhizo.cog.under.rel[13,13]; mean(as.numeric(summary.rhizo.cog.under.rel[13,1:12]))
t.test(t(summary.rhizo.cog.all.rel[17,1:12]), t(summary.rhizo.cog.under.rel[13,1:12]), paired = TRUE)
#Cell envelope
"All:"; summary.rhizo.cog.all.rel[28,13]; mean(as.numeric(summary.rhizo.cog.all.rel[28,1:12]))
"Under:"; summary.rhizo.cog.under.rel[16,13]; mean(as.numeric(summary.rhizo.cog.under.rel[16,1:12]))
t.test(t(summary.rhizo.cog.all.rel[28,1:12]), t(summary.rhizo.cog.under.rel[16,1:12]), paired = TRUE)
#Cell motility and secretion
"All:"; summary.rhizo.cog.all.rel[36,13]; mean(as.numeric(summary.rhizo.cog.all.rel[36,1:12]))
"Under:"; summary.rhizo.cog.under.rel[20,13]; mean(as.numeric(summary.rhizo.cog.under.rel[20,1:12]))
t.test(t(summary.rhizo.cog.all.rel[36,1:12]), t(summary.rhizo.cog.under.rel[20,1:12]), paired = TRUE)
#Coenzyme metabolism
"All:"; summary.rhizo.cog.all.rel[45,13]; mean(as.numeric(summary.rhizo.cog.all.rel[45,1:12]))
"Under:"; summary.rhizo.cog.under.rel[25,13]; mean(as.numeric(summary.rhizo.cog.under.rel[25,1:12]))
t.test(t(summary.rhizo.cog.all.rel[45,1:12]), t(summary.rhizo.cog.under.rel[25,1:12]), paired = TRUE)
#DNA replication
"All:"; summary.rhizo.cog.all.rel[58,13]; mean(as.numeric(summary.rhizo.cog.all.rel[58,1:12]))
"Under:"; summary.rhizo.cog.under.rel[30,13]; mean(as.numeric(summary.rhizo.cog.under.rel[30,1:12]))
t.test(t(summary.rhizo.cog.all.rel[58,1:12]), t(summary.rhizo.cog.under.rel[30,1:12]), paired = TRUE)
#Energy production and conversion
"All:"; summary.rhizo.cog.all.rel[64,13]; mean(as.numeric(summary.rhizo.cog.all.rel[64,1:12]))
"Under:"; summary.rhizo.cog.under.rel[35,13]; mean(as.numeric(summary.rhizo.cog.under.rel[35,1:12]))
t.test(t(summary.rhizo.cog.all.rel[64,1:12]), t(summary.rhizo.cog.under.rel[35,1:12]), paired = TRUE)
#Inorganic ion
"All:"; summary.rhizo.cog.all.rel[77,13]; mean(as.numeric(summary.rhizo.cog.all.rel[77,1:12]))
"Under:"; summary.rhizo.cog.under.rel[41,13]; mean(as.numeric(summary.rhizo.cog.under.rel[41,1:12]))
t.test(t(summary.rhizo.cog.all.rel[77,1:12]), t(summary.rhizo.cog.under.rel[41,1:12]), paired = TRUE)
#Intracellular trafficking
"All:"; summary.rhizo.cog.all.rel[83,13]; mean(as.numeric(summary.rhizo.cog.all.rel[83,1:12]))
"Under:"; summary.rhizo.cog.under.rel[44,13]; mean(as.numeric(summary.rhizo.cog.under.rel[44,1:12]))
t.test(t(summary.rhizo.cog.all.rel[83,1:12]), t(summary.rhizo.cog.under.rel[44,1:12]), paired = TRUE)
#Lipid
"All:"; summary.rhizo.cog.all.rel[89,13]; mean(as.numeric(summary.rhizo.cog.all.rel[89,1:12]))
"Under:"; summary.rhizo.cog.under.rel[47,13]; mean(as.numeric(summary.rhizo.cog.under.rel[47,1:12]))
t.test(t(summary.rhizo.cog.all.rel[89,1:12]), t(summary.rhizo.cog.under.rel[47,1:12]), paired = TRUE)
#Posttranslational
"All:"; summary.rhizo.cog.all.rel[102,13]; mean(as.numeric(summary.rhizo.cog.all.rel[102,1:12]))
"Under:"; summary.rhizo.cog.under.rel[53,13]; mean(as.numeric(summary.rhizo.cog.under.rel[53,1:12]))
t.test(t(summary.rhizo.cog.all.rel[102,1:12]), t(summary.rhizo.cog.under.rel[53,1:12]), paired = TRUE)
#Signal
"All:"; summary.rhizo.cog.all.rel[125,13]; mean(as.numeric(summary.rhizo.cog.all.rel[125,1:12]))
"Under:"; summary.rhizo.cog.under.rel[63,13]; mean(as.numeric(summary.rhizo.cog.under.rel[63,1:12]))
t.test(t(summary.rhizo.cog.all.rel[125,1:12]), t(summary.rhizo.cog.under.rel[63,1:12]), paired = TRUE)
#Transcription
"All:"; summary.rhizo.cog.all.rel[129,13]; mean(as.numeric(summary.rhizo.cog.all.rel[129,1:12]))
"Under:"; summary.rhizo.cog.under.rel[65,13]; mean(as.numeric(summary.rhizo.cog.under.rel[65,1:12]))
t.test(t(summary.rhizo.cog.all.rel[129,1:12]), t(summary.rhizo.cog.under.rel[65,1:12]), paired = TRUE)
#Translation
"All:"; summary.rhizo.cog.all.rel[135,13]; mean(as.numeric(summary.rhizo.cog.all.rel[135,1:12]))
"Under:"; summary.rhizo.cog.under.rel[68,13]; mean(as.numeric(summary.rhizo.cog.under.rel[68,1:12]))
t.test(t(summary.rhizo.cog.all.rel[135,1:12]), t(summary.rhizo.cog.under.rel[68,1:12]), paired = TRUE)

```

```{r anova taxa Root}
#Fungi at class
#Get table with all reps
MT.root.fun.tax.found <- MT.root.fun.tax[row.names(MT.root.fun.tax) %in% row.names(DEfound.root.fun),] #Get only DEfound
DEfound.root.fun.s <- DEfound.root.fun[order(row.names(DEfound.root.fun)),] #Sort
sum(row.names(DEfound.root.fun.s) == row.names(MT.root.fun.tax.found)) #23274 -- OK
MT.root.fun.tax.found <- data.frame(cbind(MT.root.fun.tax.found, DEfound.root.fun.s$log2))#Add column for log2
MT.root.fun.tax.found.over <- MT.root.fun.tax.found[DEfound.root.fun.s$log2 > 0,]
MT.root.fun.tax.found.under <- MT.root.fun.tax.found[DEfound.root.fun.s$log2 < 0,]
MT.root.fun.tax.found.over <- data.frame(apply(MT.root.fun.tax.found.over[,1:11], 2,      # Specify own function within apply
         function(x) as.numeric(as.character(x))), MT.root.fun.tax.found.over[,12:13])
MT.root.fun.tax.found.under <- data.frame(apply(MT.root.fun.tax.found.under[,1:11], 2,      # Specify own function within apply
         function(x) as.numeric(as.character(x))), MT.root.fun.tax.found.under[,12:13])
MT.root.fun.tax <- data.frame(apply(MT.root.fun.tax[,1:11], 2,      # Specify own function within apply
         function(x) as.numeric(as.character(x))), MT.root.fun.tax[,12])



#DEFound, only group_by phylum, keep all samples
summary.root.fun.over <- MT.root.fun.tax.found.over %>%
 group_by(V12) %>%
 summarise(
  i5_A5.14.over = sum(i5_A5.14),
  i5_A6.22.over = sum(i5_A6.22),
  i5_B5.15.over = sum(i5_B5.15),
  i5_C5.16.over = sum(i5_C5.16),
  i5_C6.24.over = sum(i5_C6.24),
  i5_D4.9.over = sum(i5_D4.9),
  i5_D5.17.over = sum(i5_D5.17),
  i5_E4.10.over = sum(i5_E4.10),
  i5_F3.3.over = sum(i5_F3.3),
  i5_F5.19.over = sum(i5_F5.19),
  i5_H3.5.over = sum(i5_H3.5),
  
 )
summary.root.fun.under <- MT.root.fun.tax.found.under %>%
 group_by(V12) %>%
 summarise(
  i5_A5.14.over = sum(i5_A5.14),
  i5_A6.22.over = sum(i5_A6.22),
  i5_B5.15.over = sum(i5_B5.15),
  i5_C5.16.over = sum(i5_C5.16),
  i5_C6.24.over = sum(i5_C6.24),
  i5_D4.9.over = sum(i5_D4.9),
  i5_D5.17.over = sum(i5_D5.17),
  i5_E4.10.over = sum(i5_E4.10),
  i5_F3.3.over = sum(i5_F3.3),
  i5_F5.19.over = sum(i5_F5.19),
  i5_H3.5.over = sum(i5_H3.5),
  
 )

summary.root.fun.all <- MT.root.fun.tax %>%
 group_by(MT.root.fun.tax...12.) %>%
 summarise(
  i5_A5.14.over = sum(i5_A5.14),
  i5_A6.22.over = sum(i5_A6.22),
  i5_B5.15.over = sum(i5_B5.15),
  i5_C5.16.over = sum(i5_C5.16),
  i5_C6.24.over = sum(i5_C6.24),
  i5_D4.9.over = sum(i5_D4.9),
  i5_D5.17.over = sum(i5_D5.17),
  i5_E4.10.over = sum(i5_E4.10),
  i5_F3.3.over = sum(i5_F3.3),
  i5_F5.19.over = sum(i5_F5.19),
  i5_H3.5.over = sum(i5_H3.5),
  
 )

#Put everything in relative
#All
summary.root.fun.all$Class <- summary.root.fun.all$MT.root.fun.tax...12.
summary.root.fun.all <- summary.root.fun.all[,-1]
summary.root.fun.all.rel <- data.frame(t(apply(summary.root.fun.all[,1:11], 1, "/", colSums(summary.root.fun.all[,1:11]))))
colSums(summary.root.fun.all.rel)
summary.root.fun.all.rel$Class <- summary.root.fun.all$Class
#Over
summary.root.fun.over$Class <- summary.root.fun.over$V12
summary.root.fun.over <- summary.root.fun.over[,-1]
summary.root.fun.over.rel <- data.frame(t(apply(summary.root.fun.over[,1:11], 1, "/", colSums(summary.root.fun.over[,1:11]))))
colSums(summary.root.fun.over.rel)
summary.root.fun.over.rel$Class <- summary.root.fun.over$Class
#under
summary.root.fun.under$Class <- summary.root.fun.under$V12
summary.root.fun.under <- summary.root.fun.under[,-1]
summary.root.fun.under.rel <- data.frame(t(apply(summary.root.fun.under[,1:11], 1, "/", colSums(summary.root.fun.under[,1:11]))))
colSums(summary.root.fun.under.rel)
summary.root.fun.under.rel$Class <- summary.root.fun.under$Class

#ALL vs. OVER
#Agaricomycetes
"All:"; summary.root.fun.all.rel[1,12]; mean(as.numeric(summary.root.fun.all.rel[1,1:11]))
"Over:"; summary.root.fun.over.rel[1,12]; mean(as.numeric(summary.root.fun.over.rel[1,1:11]))
t.test(t(summary.root.fun.all.rel[1,1:11]), t(summary.root.fun.over.rel[1,1:11]), paired = TRUE)
#Dothideomycetes
"All:"; summary.root.fun.all.rel[12,12]; mean(as.numeric(summary.root.fun.all.rel[12,1:11]))
"Over:"; summary.root.fun.over.rel[6,12]; mean(as.numeric(summary.root.fun.over.rel[6,1:11]))
t.test(t(summary.root.fun.all.rel[12,1:11]), t(summary.root.fun.over.rel[6,1:11]), paired = TRUE)
#Eurotiomycetes
"All:"; summary.root.fun.all.rel[15,12]; mean(as.numeric(summary.root.fun.all.rel[15,1:11]))
"Over:"; summary.root.fun.over.rel[9,12]; mean(as.numeric(summary.root.fun.over.rel[9,1:11]))
t.test(t(summary.root.fun.all.rel[15,1:11]), t(summary.root.fun.over.rel[9,1:11]), paired = TRUE)
#Sordariomycetes
"All:"; summary.root.fun.all.rel[40,12]; mean(as.numeric(summary.root.fun.all.rel[40,1:11]))
"Over:"; summary.root.fun.over.rel[31,12]; mean(as.numeric(summary.root.fun.over.rel[31,1:11]))
t.test(t(summary.root.fun.all.rel[40,1:11]), t(summary.root.fun.over.rel[31,1:11]), paired = TRUE)

#ALL vs. UNDER
#Agaricomycetes
"All:"; summary.root.fun.all.rel[1,12]; mean(as.numeric(summary.root.fun.all.rel[1,1:11]))
"under:"; summary.root.fun.under.rel[1,12]; mean(as.numeric(summary.root.fun.under.rel[1,1:11]))
t.test(t(summary.root.fun.all.rel[1,1:11]), t(summary.root.fun.under.rel[1,1:11]), paired = TRUE)
#Dothideomycetes
"All:"; summary.root.fun.all.rel[12,12]; mean(as.numeric(summary.root.fun.all.rel[12,1:11]))
"under:"; summary.root.fun.under.rel[4,12]; mean(as.numeric(summary.root.fun.under.rel[4,1:11]))
t.test(t(summary.root.fun.all.rel[12,1:11]), t(summary.root.fun.under.rel[4,1:11]), paired = TRUE)
#Eurotiomycetes
"All:"; summary.root.fun.all.rel[15,12]; mean(as.numeric(summary.root.fun.all.rel[15,1:11]))
"under:"; summary.root.fun.under.rel[6,12]; mean(as.numeric(summary.root.fun.under.rel[6,1:11]))
t.test(t(summary.root.fun.all.rel[15,1:11]), t(summary.root.fun.under.rel[6,1:11]), paired = TRUE)
#Sordariomycetes
"All:"; summary.root.fun.all.rel[40,12]; mean(as.numeric(summary.root.fun.all.rel[40,1:11]))
"under:"; summary.root.fun.under.rel[18,12]; mean(as.numeric(summary.root.fun.under.rel[18,1:11]))
t.test(t(summary.root.fun.all.rel[40,1:11]), t(summary.root.fun.under.rel[18,1:11]), paired = TRUE)

```

```{r anova COG Root}
#COG root fungi
#Get table with all reps
MT.root.fun.cog.found <- MT.root.fun.cog[row.names(MT.root.fun.cog) %in% row.names(DEfound.root.fun),] #Get only DEfound
DEfound.root.fun.s <- DEfound.root.fun[order(row.names(DEfound.root.fun)),] #Sort
sum(row.names(DEfound.root.fun.s) == row.names(MT.root.fun.cog.found)) #14178
MT.root.fun.cog.found <- data.frame(cbind(MT.root.fun.cog.found, DEfound.root.fun.s$log2))#Add column for log2
MT.root.fun.cog.found.over <- MT.root.fun.cog.found[DEfound.root.fun.s$log2 > 0,]
MT.root.fun.cog.found.under <- MT.root.fun.cog.found[DEfound.root.fun.s$log2 < 0,]
MT.root.fun.cog.found.over <- data.frame(apply(MT.root.fun.cog.found.over[,1:11], 2,      # Specify own function within apply
         function(x) as.numeric(as.character(x))), MT.root.fun.cog.found.over[,12:13])
MT.root.fun.cog.found.under <- data.frame(apply(MT.root.fun.cog.found.under[,1:11], 2,      # Specify own function within apply
         function(x) as.numeric(as.character(x))), MT.root.fun.cog.found.under[,12:13])
MT.root.fun.cog <- data.frame(apply(MT.root.fun.cog[,1:11], 2,      # Specify own function within apply
         function(x) as.numeric(as.character(x))), MT.root.fun.cog[,12])



#DEFound, only group_by phylum, keep all samples
summary.root.cog.over <- MT.root.fun.cog.found.over %>%
 group_by(V12) %>%
 summarise(
  i5_A5.14.over = sum(i5_A5.14),
  i5_A6.22.over = sum(i5_A6.22),
  i5_B5.15.over = sum(i5_B5.15),
  i5_C5.16.over = sum(i5_C5.16),
  i5_C6.24.over = sum(i5_C6.24),
  i5_D4.9.over = sum(i5_D4.9),
  i5_D5.17.over = sum(i5_D5.17),
  i5_E4.10.over = sum(i5_E4.10),
  i5_F3.3.over = sum(i5_F3.3),
  i5_F5.19.over = sum(i5_F5.19),
  i5_H3.5.over = sum(i5_H3.5),
  
 )
summary.root.cog.under <- MT.root.fun.cog.found.under %>%
 group_by(V12) %>%
 summarise(
   i5_A5.14.under = sum(i5_A5.14),
  i5_A6.22.under = sum(i5_A6.22),
  i5_B5.15.under = sum(i5_B5.15),
  i5_C5.16.under = sum(i5_C5.16),
  i5_C6.24.under = sum(i5_C6.24),
  i5_D4.9.under = sum(i5_D4.9),
  i5_D5.17.under = sum(i5_D5.17),
  i5_E4.10.under = sum(i5_E4.10),
  i5_F3.3.under = sum(i5_F3.3),
  i5_F5.19.under = sum(i5_F5.19),
  i5_H3.5.under = sum(i5_H3.5),
  
 )

summary.root.cog.all <- MT.root.fun.cog %>%
 group_by(MT.root.fun.cog...12.) %>%
 summarise(
   i5_A5.14.all = sum(i5_A5.14),
  i5_A6.22.all = sum(i5_A6.22),
  i5_B5.15.all = sum(i5_B5.15),
  i5_C5.16.all = sum(i5_C5.16),
  i5_C6.24.all = sum(i5_C6.24),
  i5_D4.9.all = sum(i5_D4.9),
  i5_D5.17.all = sum(i5_D5.17),
  i5_E4.10.all = sum(i5_E4.10),
  i5_F3.3.all = sum(i5_F3.3),
  i5_F5.19.all = sum(i5_F5.19),
  i5_H3.5.all = sum(i5_H3.5),
 )

#Put everything in relative
#All
summary.root.cog.all$COG_cat <- summary.root.cog.all$MT.root.fun.cog...12.
summary.root.cog.all <- summary.root.cog.all[,-1]
summary.root.cog.all.rel <- data.frame(t(apply(summary.root.cog.all[,1:11], 1, "/", colSums(summary.root.cog.all[,1:11]))))
colSums(summary.root.cog.all.rel)
summary.root.cog.all.rel$COG_cat <- summary.root.cog.all$COG_cat
#Over
summary.root.cog.over$COG_cat <- summary.root.cog.over$V12
summary.root.cog.over <- summary.root.cog.over[,-1]
summary.root.cog.over.rel <- data.frame(t(apply(summary.root.cog.over[,1:11], 1, "/", colSums(summary.root.cog.over[,1:11]))))
colSums(summary.root.cog.over.rel)
summary.root.cog.over.rel$COG_cat <- summary.root.cog.over$COG_cat
#under
summary.root.cog.under$COG_cat <- summary.root.cog.under$V12
summary.root.cog.under <- summary.root.cog.under[,-1]
summary.root.cog.under.rel <- data.frame(t(apply(summary.root.cog.under[,1:11], 1, "/", colSums(summary.root.cog.under[,1:11]))))
colSums(summary.root.cog.under.rel)
summary.root.cog.under.rel$COG_cat <- summary.root.cog.under$COG_cat

#ALL vs. OVER
#AA trans and metab
"All:"; summary.root.cog.all.rel[3,12]; mean(as.numeric(summary.root.cog.all.rel[3,1:11]))
"Over:"; summary.root.cog.over.rel[2,12]; mean(as.numeric(summary.root.cog.over.rel[2,1:11]))
t.test(t(summary.root.cog.all.rel[3,1:11]), t(summary.root.cog.over.rel[2,1:11]), paired = TRUE)
#Carb trans and metabolism
"All:"; summary.root.cog.all.rel[12,12]; mean(as.numeric(summary.root.cog.all.rel[12,1:11]))
"Over:"; summary.root.cog.over.rel[11,12]; mean(as.numeric(summary.root.cog.over.rel[11,1:11]))
t.test(t(summary.root.cog.all.rel[12,1:11]), t(summary.root.cog.over.rel[11,1:11]), paired = TRUE)
#Cell envelope
"All:"; summary.root.cog.all.rel[29,12]; mean(as.numeric(summary.root.cog.all.rel[29,1:11]))
"Over:"; summary.root.cog.over.rel[27,12]; mean(as.numeric(summary.root.cog.over.rel[27,1:11]))
t.test(t(summary.root.cog.all.rel[29,1:11]), t(summary.root.cog.over.rel[27,1:11]), paired = TRUE)
#Cell motility and secretion
"All:"; summary.root.cog.all.rel[34,12]; mean(as.numeric(summary.root.cog.all.rel[34,1:11]))
"Over:"; summary.root.cog.over.rel[32,12]; mean(as.numeric(summary.root.cog.over.rel[32,1:11]))
t.test(t(summary.root.cog.all.rel[34,1:11]), t(summary.root.cog.over.rel[32,1:11]), paired = TRUE)
#Coenzyme metabolism
"All:"; summary.root.cog.all.rel[40,12]; mean(as.numeric(summary.root.cog.all.rel[40,1:11]))
"Over:"; summary.root.cog.over.rel[37,12]; mean(as.numeric(summary.root.cog.over.rel[37,1:11]))
t.test(t(summary.root.cog.all.rel[40,1:11]), t(summary.root.cog.over.rel[37,1:11]), paired = TRUE)
#DNA replication
"All:"; summary.root.cog.all.rel[46,12]; mean(as.numeric(summary.root.cog.all.rel[46,1:11]))
"Over:"; summary.root.cog.over.rel[43,12]; mean(as.numeric(summary.root.cog.over.rel[43,1:11]))
t.test(t(summary.root.cog.all.rel[46,1:11]), t(summary.root.cog.over.rel[43,1:11]), paired = TRUE)
#Energy production and conversion
"All:"; summary.root.cog.all.rel[52,12]; mean(as.numeric(summary.root.cog.all.rel[52,1:11]))
"Over:"; summary.root.cog.over.rel[46,12]; mean(as.numeric(summary.root.cog.over.rel[46,1:11]))
t.test(t(summary.root.cog.all.rel[52,1:11]), t(summary.root.cog.over.rel[46,1:11]), paired = TRUE)
#Inorganic ion
"All:"; summary.root.cog.all.rel[61,12]; mean(as.numeric(summary.root.cog.all.rel[61,1:11]))
"Over:"; summary.root.cog.over.rel[55,12]; mean(as.numeric(summary.root.cog.over.rel[55,1:11]))
t.test(t(summary.root.cog.all.rel[61,1:11]), t(summary.root.cog.over.rel[55,1:11]), paired = TRUE)
#Intracellular trafficking
"All:"; summary.root.cog.all.rel[64,12]; mean(as.numeric(summary.root.cog.all.rel[64,1:11]))
"Over:"; summary.root.cog.over.rel[57,12]; mean(as.numeric(summary.root.cog.over.rel[57,1:11]))
t.test(t(summary.root.cog.all.rel[64,1:11]), t(summary.root.cog.over.rel[57,1:11]), paired = TRUE)
#Lipid
"All:"; summary.root.cog.all.rel[69,12]; mean(as.numeric(summary.root.cog.all.rel[69,1:11]))
"Over:"; summary.root.cog.over.rel[61,12]; mean(as.numeric(summary.root.cog.over.rel[61,1:11]))
t.test(t(summary.root.cog.all.rel[69,1:11]), t(summary.root.cog.over.rel[61,1:11]), paired = TRUE)
#Posttranslational
"All:"; summary.root.cog.all.rel[83,12]; mean(as.numeric(summary.root.cog.all.rel[83,1:11]))
"Over:"; summary.root.cog.over.rel[75,12]; mean(as.numeric(summary.root.cog.over.rel[75,1:11]))
t.test(t(summary.root.cog.all.rel[83,1:11]), t(summary.root.cog.over.rel[75,1:11]), paired = TRUE)
#Signal
"All:"; summary.root.cog.all.rel[96,12]; mean(as.numeric(summary.root.cog.all.rel[96,1:11]))
"Over:"; summary.root.cog.over.rel[84,12]; mean(as.numeric(summary.root.cog.over.rel[84,1:11]))
t.test(t(summary.root.cog.all.rel[96,1:11]), t(summary.root.cog.over.rel[84,1:11]), paired = TRUE)
#Transcription
"All:"; summary.root.cog.all.rel[104,12]; mean(as.numeric(summary.root.cog.all.rel[104,1:11]))
"Over:"; summary.root.cog.over.rel[92,12]; mean(as.numeric(summary.root.cog.over.rel[92,1:11]))
t.test(t(summary.root.cog.all.rel[104,1:11]), t(summary.root.cog.over.rel[92,1:11]), paired = TRUE)
#Translation
"All:"; summary.root.cog.all.rel[111,12]; mean(as.numeric(summary.root.cog.all.rel[111,1:11]))
"Over:"; summary.root.cog.over.rel[99,12]; mean(as.numeric(summary.root.cog.over.rel[99,1:11]))
t.test(t(summary.root.cog.all.rel[111,1:11]), t(summary.root.cog.over.rel[99,1:11]), paired = TRUE)

#ALL vs. UNDER
#AA trans and metab
"All:"; summary.root.cog.all.rel[3,12]; mean(as.numeric(summary.root.cog.all.rel[3,1:11]))
"under:"; summary.root.cog.under.rel[1,12]; mean(as.numeric(summary.root.cog.under.rel[1,1:11]))
t.test(t(summary.root.cog.all.rel[3,1:11]), t(summary.root.cog.under.rel[1,1:11]), paired = TRUE)
#Carb trans and metabolism
"All:"; summary.root.cog.all.rel[12,12]; mean(as.numeric(summary.root.cog.all.rel[12,1:11]))
"under:"; summary.root.cog.under.rel[2,12]; mean(as.numeric(summary.root.cog.under.rel[2,1:11]))
t.test(t(summary.root.cog.all.rel[12,1:11]), t(summary.root.cog.under.rel[2,1:11]), paired = TRUE)
#Cell envelope
#"All:"; summary.root.cog.all.rel[29,12]; mean(as.numeric(summary.root.cog.all.rel[29,1:11]))
#"under:"; summary.root.cog.under.rel[3,12]; mean(as.numeric(summary.root.cog.under.rel[3,1:11]))
#t.test(t(summary.root.cog.all.rel[29,1:11]), t(summary.root.cog.under.rel[3,1:11]), paired = TRUE)
#Cell motility and secretion
#"All:"; summary.root.cog.all.rel[34,12]; mean(as.numeric(summary.root.cog.all.rel[34,1:11]))
#"under:"; summary.root.cog.under.rel[32,12]; mean(as.numeric(summary.root.cog.under.rel[32,1:11]))
#t.test(t(summary.root.cog.all.rel[34,1:11]), t(summary.root.cog.under.rel[32,1:11]), paired = TRUE)
#Coenzyme metabolism
#"All:"; summary.root.cog.all.rel[40,12]; mean(as.numeric(summary.root.cog.all.rel[40,1:11]))
#"under:"; summary.root.cog.under.rel[37,12]; mean(as.numeric(summary.root.cog.under.rel[37,1:11]))
#t.test(t(summary.root.cog.all.rel[40,1:11]), t(summary.root.cog.under.rel[37,1:11]), paired = TRUE)
#DNA replication
#"All:"; summary.root.cog.all.rel[46,12]; mean(as.numeric(summary.root.cog.all.rel[46,1:11]))
#"under:"; summary.root.cog.under.rel[43,12]; mean(as.numeric(summary.root.cog.under.rel[43,1:11]))
#t.test(t(summary.root.cog.all.rel[46,1:11]), t(summary.root.cog.under.rel[43,1:11]), paired = TRUE)
#Energy production and conversion
"All:"; summary.root.cog.all.rel[52,12]; mean(as.numeric(summary.root.cog.all.rel[52,1:11]))
"under:"; summary.root.cog.under.rel[7,12]; mean(as.numeric(summary.root.cog.under.rel[7,1:11]))
t.test(t(summary.root.cog.all.rel[52,1:11]), t(summary.root.cog.under.rel[7,1:11]), paired = TRUE)
#Inorganic ion
"All:"; summary.root.cog.all.rel[61,12]; mean(as.numeric(summary.root.cog.all.rel[61,1:11]))
"under:"; summary.root.cog.under.rel[11,12]; mean(as.numeric(summary.root.cog.under.rel[11,1:11]))
t.test(t(summary.root.cog.all.rel[61,1:11]), t(summary.root.cog.under.rel[11,1:11]), paired = TRUE)
#Intracellular trafficking
#"All:"; summary.root.cog.all.rel[64,12]; mean(as.numeric(summary.root.cog.all.rel[64,1:11]))
#"under:"; summary.root.cog.under.rel[57,12]; mean(as.numeric(summary.root.cog.under.rel[57,1:11]))
#t.test(t(summary.root.cog.all.rel[64,1:11]), t(summary.root.cog.under.rel[57,1:11]), paired = TRUE)
#Lipid
"All:"; summary.root.cog.all.rel[69,12]; mean(as.numeric(summary.root.cog.all.rel[69,1:11]))
"under:"; summary.root.cog.under.rel[12,12]; mean(as.numeric(summary.root.cog.under.rel[12,1:11]))
t.test(t(summary.root.cog.all.rel[69,1:11]), t(summary.root.cog.under.rel[12,1:11]), paired = TRUE)
#Posttranslational
"All:"; summary.root.cog.all.rel[83,12]; mean(as.numeric(summary.root.cog.all.rel[83,1:11]))
"under:"; summary.root.cog.under.rel[14,12]; mean(as.numeric(summary.root.cog.under.rel[14,1:11]))
t.test(t(summary.root.cog.all.rel[83,1:11]), t(summary.root.cog.under.rel[14,1:11]), paired = TRUE)
#Signal
#"All:"; summary.root.cog.all.rel[96,12]; mean(as.numeric(summary.root.cog.all.rel[96,1:11]))
#"under:"; summary.root.cog.under.rel[84,12]; mean(as.numeric(summary.root.cog.under.rel[84,1:11]))
#t.test(t(summary.root.cog.all.rel[96,1:11]), t(summary.root.cog.under.rel[84,1:11]), paired = TRUE)
#Transcription
"All:"; summary.root.cog.all.rel[104,12]; mean(as.numeric(summary.root.cog.all.rel[104,1:11]))
"under:"; summary.root.cog.under.rel[17,12]; mean(as.numeric(summary.root.cog.under.rel[17,1:11]))
t.test(t(summary.root.cog.all.rel[104,1:11]), t(summary.root.cog.under.rel[17,1:11]), paired = TRUE)
#Translation
"All:"; summary.root.cog.all.rel[111,12]; mean(as.numeric(summary.root.cog.all.rel[111,1:11]))
"under:"; summary.root.cog.under.rel[18,12]; mean(as.numeric(summary.root.cog.under.rel[18,1:11]))
t.test(t(summary.root.cog.all.rel[111,1:11]), t(summary.root.cog.under.rel[18,1:11]), paired = TRUE)

```

```{r soil water content}
SWC$PPT <- factor(SWC$PPT, c("25", "50", "75", "100"))#Reorder manually, and as factor
min(SWC$SWC)
max(SWC$SWC)

#ANOVA
bartlett.test(SWC$SWC, SWC$PPT) #P = 0.5661, OK
shapiro.test(SWC$SWC) #P = 0.7003, OK
anova.SWC <- aov(SWC~PPT+Block, data = SWC)
summary(anova.SWC) #P = 0.000367 for PPT and P = 0.049648 for Blocks
TukeyHSD(anova.SWC) #25 = a, 50 = ab, 75 = bc, 100 = c
tukey.SWC <- data.frame(letters = c("a", "ab", "bc", "c"), PPT = c("25","50", "75", "100"), position = c(25,25,25,25))#create dataframe with tukey results

#Boxplot
palette(c("red", "orange","purple", "blue"))
SWC.plot <- ggplot(SWC, aes(y = SWC, x = PPT, fill = PPT)) + 
  geom_boxplot(outlier.color = NA, outlier.size = 0, outlier.shape = NA) +
  geom_point(position = position_jitterdodge()) +
  ylab("Soil water content (%)") + 
  scale_fill_manual(values = palette()) +
  theme_bw() +
  theme(legend.position = "none") +
  scale_x_discrete(name = "Precipitation treatments") +
  geom_text(data = tukey.SWC, aes(x = PPT, y = position, label = letters)) #Add letters from tukey
  
SWC.plot

```

```{r create figures}
#Figure 1
fig1 <- ggarrange(SWC.plot, stack.stats.all,stack.stats.DE, labels = c("A", "B", "C"))
fig1

ggsave(file = "../output/figs/fig1.eps", fig1, width = 7, height = 7, units = "in")
ggsave(file = "../output/figs/fig1.tiff", fig1, width = 7, height = 7, units = "in", dpi = 600, compression = "lzw")

#Figure 2
fig2 <- ggarrange(volcano.root, volcano.rhizo, labels = c("A", "B"))
fig2

ggsave(file = "../output/figs/fig2.eps", fig2, width = 7, height = 7, units = "in")
ggsave(file = "../output/figs/fig2.tiff", fig2, width = 7, height = 7, units = "in", dpi = 600, compression = "lzw")

#Figure 3
fig3 <- ggarrange(stack.root.class, stack.phyla.rhizo, labels = c("A", "B"), legend = "right" )
fig3

ggsave(file = "../output/figs/fig3.eps", fig3, width = 7, height = 7, units = "in")
ggsave(file = "../output/figs/fig3.tiff", fig3, width = 7, height = 7, units = "in", dpi = 600, compression = "lzw")

#Figure 4
fig4 <- ggarrange(stack.COG.root, stack.COG.rhizo, labels = c("A", "B"), common.legend = TRUE, legend = "right")
fig4

ggsave(file = "../output/figs/fig4.eps", fig4, width = 7, height = 7, units = "in")
ggsave(file = "../output/figs/fig4.tiff", fig4, width = 7, height = 7, units = "in", dpi = 600, compression = "lzw")

```


